app/src/main/cpp/native-lib.cpp

```diff
@@ -36,83 +30,175 @@ namespace
+    auto sdk = new BanubaSdkManager(path, token, width, height);
+    return (jlong)sdk;
 }
 
+/* kotlin BanubaSdk::destroyOffscreenEffectPlayer */
 extern "C" JNIEXPORT void JNICALL
-Java_com_banuba_sdk_example_quickstart_1cpp_BanubaSdk_destroyEffectPlayer(JNIEnv* env, jobject thiz, jlong effect_player)
+Java_com_banuba_sdk_example_quickstart_1c_1api_BanubaSdk_destroyOffscreenEffectPlayer(JNIEnv* env, jobject thiz, jlong jsdk)
 {
-    auto sdk = (BanubaSdkManager*) effect_player;
+    auto sdk = reinterpret_cast<BanubaSdkManager*>(jsdk);
     delete sdk;
 }
 
+
+// oep: Long, image: ByteBuffer, width: Int, height: Int, pxfmt: PixelFormat, callback:
+
+// callback: (image: ByteArray, width: Int, height: Int, pxfmt: PixelFormat) -> Unit
+/* kotlin BanubaSdk::processImageAsync */
 extern "C" JNIEXPORT void JNICALL
-Java_com_banuba_sdk_example_quickstart_1cpp_BanubaSdk_loadEffect(JNIEnv* env, jobject thiz, jlong effect_player, jstring name)
+Java_com_banuba_sdk_example_quickstart_1c_1api_BanubaSdk_processImageAsync(JNIEnv* env, jobject thiz, jlong jsdk, jobject jimage, jint jwidth, jint jheight)
 {
-    auto sdk = (BanubaSdkManager*) effect_player;
-    auto effect_name = jstring2string(env, name);
-
-    sdk->oep->load_effect(effect_name);
+    auto oep = reinterpret_cast<BanubaSdkManager*>(jsdk)->oep;
+    //pixel_buffer_sptr s;
+    auto* input_image_data = static_cast<uint8_t*>(env->GetDirectBufferAddress(jimage));
+    auto width = static_cast<int32_t>(jwidth);
+    auto height = static_cast<int32_t>(jheight);
+
+    // create an image
+    using ns_pb = bnb::oep::interfaces::pixel_buffer;
+    int32_t stride = width * 4;
+    ns_pb::plane_sptr rgba_plane_data(input_image_data, [](uint8_t*) { /* DO NOTHING */ });
+    ns_pb::plane_data rgba_plane{rgba_plane_data, static_cast<size_t>(stride * height), stride};
+    std::vector<ns_pb::plane_data> planes{rgba_plane};
+    auto pb_image = ns_pb::create(planes, bnb::oep::interfaces::image_format::bpc8_rgba, width, height, [](auto* pb) {});
+
+    JavaVM* jvm;
+    env->GetJavaVM(&jvm);
+    jobject this_ref = env->NewGlobalRef(thiz);
+
+    // Callback for received pixel buffer from the offscreen effect player
+    auto get_pixel_buffer_callback = [this_ref, jvm](image_processing_result_sptr result) {
+        if (result != nullptr) {
+            // Callback for update data in render thread
+            auto get_image_callback = [this_ref, jvm](pixel_buffer_sptr image) {
+                if (image == nullptr) {
+                    return;
+                }
+                JNIEnv* env = nullptr;
+
+                // double check it's all ok
+                int getEnvStat = jvm->GetEnv((void**)&env, JNI_VERSION_1_6);
+                if (getEnvStat == JNI_EDETACHED) {
+                    std::cout << "GetEnv: not attached" << std::endl;
+                    if (jvm->AttachCurrentThread((JNIEnv**)&env, nullptr) != JNI_OK) {
+                        std::cout << "Failed to attach" << std::endl;
```

Comment:
cout doesn't work in Android apps, use Android logging instead, see https://stackoverflow.com/questions/12152555/how-to-print-log-in-android-c-file

Reply:
It can be replaced as an exception anyway it is a critical error.